src/features/CodeLinkFeature.ts

```diff
@@ -278,16 +370,23 @@ class CodePosition {
 			};
 		}
 
+		let rangeObj = {};
+		if (this.range) {
+			rangeObj = {
+				start: toPosition(this.range.start),
+				end: toPosition(this.range.end),
+			};
+		} else if (this.symbol) {
+			rangeObj = {
+				symbol: this.symbol
+			};
+		}
+
 		const data: Data = {
 			path: path
 				.relative(relativeTo.fsPath, this.uri.fsPath)
 				.replace(/\\/g, "/"),
```

Comment:
In case of a symbol, is the path neccessary here?

Reply:
Ahh, the symbol is still relative to the document. Can we make it global? Otherwise, renaming a file will break the symbol reference.

Reply:
Sure, I'll add a way to link it to a top level workspace symbol in the hierarchy. However, we should still allow linking to files as some language servers don't expose workspace symbols in a hierarchy.